Ryujinx/Ui/RendererWidgetBase.cs

```diff
@@ -612,6 +614,36 @@ private bool UpdateFrame()
                         Device.SetVolume(0);
                     }
                 }
+                
+                if (currentHotkeyState.HasFlag(KeyboardHotkeyState.ResScaleUp) &&
+                    !_prevHotkeyState.HasFlag(KeyboardHotkeyState.ResScaleUp))
+                {
+                    _currentScale = Graphics.Gpu.GraphicsConfig.ResScale;
+
+                    // Cap this hotkey at 4x scaling.
+                    if (_currentScale < 4f)
+                    {
+                        // If custom scaling used then first round the float down then increase.
+                        float _scaleRound = (float)Math.Floor(_currentScale);
+                        _scaleRound += 1f;
+                        Graphics.Gpu.GraphicsConfig.ResScale = _scaleRound;
+                    }
+                }
+
+                if (currentHotkeyState.HasFlag(KeyboardHotkeyState.ResScaleDown) &&
+                    !_prevHotkeyState.HasFlag(KeyboardHotkeyState.ResScaleDown))
+                {
+                    _currentScale = Graphics.Gpu.GraphicsConfig.ResScale;
+
+                    // Cap this hotkey at 1x scaling.
+                    if(_currentScale > 1f)
+                    {
+                        // If custom scaling used then first round the float up then decrease.
+                        float _scaleRound = (float)Math.Ceiling(_currentScale);
+                        _scaleRound -= 1f;
+                        Graphics.Gpu.GraphicsConfig.ResScale = _scaleRound;
+                    }
+                }
```

Comment:
It would simplify all this if you just make the value wrap around, so if you keep pressing Page Up, it just goes 1, 2, 3, 4, 1, 2, 3, 4.

You can then just change all of this code to:

```suggestion
                const int MaxResolutionScale = 4;

                if (currentHotkeyState.HasFlag(KeyboardHotkeyState.ResScaleUp) &&
                    !_prevHotkeyState.HasFlag(KeyboardHotkeyState.ResScaleUp))
                {
                    Graphics.Gpu.GraphicsConfig.ResScale = Graphics.Gpu.GraphicsConfig.ResScale % MaxResolutionScale + 1;
                }

                if (currentHotkeyState.HasFlag(KeyboardHotkeyState.ResScaleDown) &&
                    !_prevHotkeyState.HasFlag(KeyboardHotkeyState.ResScaleDown))
                {
                    Graphics.Gpu.GraphicsConfig.ResScale = 
                        (MaxResolutionScale + Graphics.Gpu.GraphicsConfig.ResScale - 2)  % MaxResolutionScale + 1;
                }
```